### PR TITLE
Feature/check for redmod on start

### PR DESCRIPTION
**Pull request by E1337Kat**
We should make sure that users have the REDmod DLC installed when starting the extension. In theory we would have mod types to determine if the prompt is needed. but this is probably good enough for now.

### CODE REVIEW

File: src/extension.ts [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    await Promise.all([redLauncherPath, redModPath].map(async (file) => fs.statAsync(file)));
+    return;
+  } catch (err) {
+    // If this fails, then we know REDmod isn't installed properly.
+    vortexApi.log(`debug`, `REDmod not found for Cyberpunk 2077`, err);
+  }
+
+  // Determine which game store this is from, so we can recommend the correct process.
+  const game = await vortexUtil.GameStoreHelper.findByAppId([GOGAPP_ID, STEAMAPP_ID, EPICAPP_ID]);
+  if (game?.path !== discovery.path) vortexApi.log(`warn`, `Cyberpunk discovery doesn't match auto-detected path`, { discovery: discovery.path, autoDetect: game.path });
```
Comment (effs) [inline src/extension.ts:68]: I can never trust a no-block if!

```suggestion
  if (game?.path !== discovery.path) {
    vortexApi.log(`warn`, `Cyberpunk discovery doesn't match auto-detected path`, { discovery: discovery.path, autoDetect: game.path });
  }
```

File: src/extension.ts [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+  const gameStoreData = {
+    epic: {
+      name: `Epic Games Store`,
+      url: `https://store.epicgames.com/en-US/p/cyberpunk-2077`,
+    },
+    steam: {
+      name: `Steam`,
+      url: `steam://run/2060310`,
+    //   url: `https://store.steampowered.com/app/2060310/Cyberpunk_2077_REDmod/`,
```
Comment (effs) [inline src/extension.ts:25]: Is this needed for some context? If not, nix it
```suggestion
```

File: src/extension.ts [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+  VortexApi,
+  VortexDiscoveryResult,
+  VortexExtensionContext,
+  vortexUtil,
+} from "./vortex-wrapper";
+
+// This function runs on starting up Vortex or switching to Cyberpunk as the active game. This may need to be converted to a test, but the UI for tests is less flexible.
+
+const getREDmodetails = (id: string): { name: string, url: string } => {
+  if (!id || ![`epic`, `gog`, `steam`].includes(id)) return { name: undefined, url: `https://www.cyberpunk.net/en/modding-support` };
```
Comment (effs) [inline src/extension.ts:15]: I 💜 self-documenting intermediates

```suggestion
  const genericHelpUrl = `https://www.cyberpunk.net/en/modding-support`;
  
  const isRedModSupportingGamePlatform = [`epic`, `gog`, `steam`].includes(id);
  
  if (!isRedModSupportingGamePlatform) {
    return { name: undefined, url: genericHelpUrl };
  }
```

File: src/index.ts [hunk 4]
```diff
@@ -2,25 +2,39 @@ import path from "path";
...
+      path.join(`CyberCAT`, `CP2077SaveEditor.exe`),
+      path.join(`CyberCAT`, `licenses`, `CyberCAT.Core.LICENSE.txt`),
     ],
     defaultPrimary: true,
     shell: false,
     relative: true,
   },
+  {
+    id: `REDLauncher`,
+    name: `REDLauncher`,
```
Comment (effs) [inline src/index.ts:32]: Should these be called _pre_ launcher also? Was that the term used in the docs?

**Review by effs — APPROVED** [review]
Really nice, thank you!!

Left a couple non-essential suggestions but my only real concern here is the name `extension.ts` — that's what `index.ts` is, and it’s not quite clear to me what the responsibility of the new file is. We can ponder on that later, though. It might make sense to extract the REDmod stuff into its own file like the Vortex wrapper maybe?

File: src/extension.ts (hunk 1 above)
Comment (E1337Kat) [reply src/extension.ts:68]: tbh, I didn't even know it was possible before Picky's thing. I'll probably change it, ya.

File: src/extension.ts (hunk 2 above)
Comment (E1337Kat) [reply src/extension.ts:25]: I think it is nice to see the context.... but steam should be fine to handle it, ya?

File: src/index.ts (hunk 4 above)
Comment (E1337Kat) [reply src/index.ts:32]: I'm not sure???

**Comment by E1337Kat** [comment]
Regarding the file name, I am not like set on it. it was camel case, but I wanted it not to be. Maybe `redmodding.ts`?

**Comment by E1337Kat** [comment]
also, real tired of these tests not passing :(

File: src/index.ts (hunk 4 above)
Comment (Pickysaurus) [reply src/index.ts:32]: CDPR call it REDlauncher when we spoke to them.

**Pull request merged**